pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -96,15 +98,15 @@ func (i *instantSplitter) copyWithEmbeddedExpr(embeddedExpr *parser.AggregateExp
 }
 
 // mapAggregatorExpr maps vector aggregator expression expr
-func (i *instantSplitter) mapAggregatorExpr(expr *parser.AggregateExpr, stats *MapperStats) (mapped parser.Expr, finished bool, err error) {
+func (i *instantSplitter) mapAggregatorExpr(expr *parser.AggregateExpr) (mapped parser.Expr, finished bool, err error) {
 	var mappedExpr parser.Expr
 
 	// If the outerAggregationExpr is not set, update it.
 	// Note: vector aggregators avg, count and topk are supported but not splittable, so cannot be sent downstream.
 	if i.outerAggregationExpr == nil && splittableVectorAggregators[expr.Op] {
-		mappedExpr, finished, err = NewASTExprMapper(i.copyWithEmbeddedExpr(expr)).MapExpr(expr.Expr, stats)
+		mappedExpr, finished, err = NewASTExprMapper(i.copyWithEmbeddedExpr(expr)).MapExpr(expr.Expr)
```

Comment:
Note to reviewers: here it works because `copyWithEmbeddedExpr()` keeps the same `stats` reference in the copied struct.